### ISSUE TITLE
Breaks when command contains `&&`?

### ISSUE THREAD

**Issue by ljharb** [kimmobrunfeldt/concurrently#31]
If you clone https://github.com/airbnb/mocha-wrap and check out the `concurrently` branch, you'll see that `npm run tests-only` [fails](https://travis-ci.org/airbnb/mocha-wrap/builds/117226816) with, for example, "Error: cannot resolve path (or pattern) '&&'".

I would assume that `concurrently 'foo' 'bar'` would work for any `foo` or `bar`, even if it was multiple commands strung together serially.

The same tests [work fine](https://travis-ci.org/airbnb/mocha-wrap/builds/117226775) on the `master` branch with `parallelshell`.

**Comment by rweng** [comment]
In some cases, running things in parallel is not enough. e.g, i fire up a docker container and then I want to run some integration tests against it. Thus, I thought I'd just sleep a short time before before starting the tests, when they are done `-k` should take care of stopping the container.

```bash
RW-MBP-15:  ~
→ sleep 1 && echo hello
hello

RW-MBP-15:  ~
→ concurrently 'sleep 1 && echo hello'
[0] usage: sleep seconds
[0] sleep 1 && echo hello exited with code 1
```

**Comment by jonkri** [comment]
+1

Separating different commands with semi-colon doesn't seem to work either.

**Comment by kimmobrunfeldt** [comment]
Yeah this sucks and should be fixed. We should have all shell capabilities in the commands but haven't found a good way to do it well cross-platform. I'll try to peak what parallelshell does. If someone has a fix, I'll merge a PR.

**Comment by kimmobrunfeldt** [comment]
Could you test if this has been fixed by installing 3.0.0-dev version: `npm i -g concurrently@3.0.0-dev` ? 
It should be better now because it is using https://github.com/kimmobrunfeldt/spawn-default-shell for spawning shell.

**Comment by ljharb** [comment]
@kimmobrunfeldt `concurrently 'echo 1 && exit 1' 'echo 2 && exit 2'` is still incorrectly passing with 3.0.0-dev - that seems to be a simple test case.

**Comment by kimmobrunfeldt** [comment]
What do you mean by "is still incorrectly passing"?

To me this behaviour seems to be correct:

```bash
➜  concurrently git:(master) ✗ concurrently 'echo 1 && exit 1' 'echo 2 && exit 2'
[0] 1
[1] 2
[1] echo 2 && exit 2 exited with code 2
[0] echo 1 && exit 1 exited with code 1
➜  concurrently git:(master) ✗ echo $?
1
```

concurrently returns with exit code 1.

**Comment by kimmobrunfeldt** [comment]
@rweng also now `concurrently 'sleep 1 && echo hello'` works fine:

```bash
➜  concurrently git:(master) ✗ concurrently 'sleep 1 && echo hello'
[0] hello
[0] sleep 1 && echo hello exited with code 0
```

**Comment by kimmobrunfeldt** [comment]
Are you sure you uninstalled the old one and used 3.0.0-dev version? Also are you sure that your platform shell is compatible with those commands? Which OS are you running? E.g in Macs, spawn-default-shell will use the shell defined in SHELL env or /bin/bash -c "..." to run the given commands. I also made a test case for this - it passed on my Mac but I can double check on Travis CI too.

**Comment by ljharb** [comment]
It's a Mac, stock, using bash. I'll try again this evening, just in case something went wrong with my install.

**Comment by kimmobrunfeldt** [comment]
Ok I'll commit the test case to master later today too.

**Comment by kimmobrunfeldt** [comment]
The test case is here https://github.com/kimmobrunfeldt/concurrently/blob/master/test/test-functional.js#L78 and it is passing in Travis: https://travis-ci.org/kimmobrunfeldt/concurrently

**Comment by ljharb** [comment]
Hmm, I must have made a mistake, because I definitely have 3.0.0-dev installed, didn't change anything, and got the proper exit codes. Thanks!

**Comment by kimmobrunfeldt** [comment]
These issues should be now fixed in 3.0.0-rc1. You can test by installing it: `npm i -g concurrently@3.0.0-rc1`. Please open a new issue to concurrently repo if this is not the case. I'll release the 3.0.0 soonish.
